addons/calendar/models/mail_activity_mixin.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class MailActivityMixin(models.AbstractModel):
+    _inherit = 'mail.activity.mixin'
+
+    activity_calendar_event_id = fields.Many2one(
+        'calendar.event', "Next Activity Calendar Event", compute='_compute_activity_calendar_event_id',
+        search='_search_activity_calendar_event_id', groups="base.group_user")
+
+    @api.depends('activity_ids.calendar_event_id')
+    def _compute_activity_calendar_event_id(self):
+        for record in self:
+            record.activity_calendar_event_id = record.activity_ids[:1].calendar_event_id
+
+    @api.model
+    def _search_activity_calendar_event_id(self, operator, operand):
+        return [('activity_ids.calendar_event_id', operator, operand)]
```

Comment:
Actually do we really want to support search on it ? Does not seem to be available in search view.